### PR TITLE
fix sparse categorical acc

### PR DESCRIPTION
**Pull request by roywei**
### Summary

1. sparse categorical acc should have the same result as categorical acc.

For example, with 3 classes, given `sparse_true_label = [0, 1, 1]`, it's equivalent in categorical labels is `dense_true_label = [[1, 0, 0], [0, 1, 0], [0, 1, 0]]`. With the same predictions 
```
pred = [[0.7, 0.2, 0.1], 
[0.1, 0.1, 0.8], 
[0.2, 0.6, 0.2]]
```
They should produce the same acc which is `[1, 0 ,1]`

2. Not sure why `max` is used, but it should directly compare with `y_true`
3. Added unit test to test correctness

### PR Overview

- [x] This PR requires new unit tests [y] (make sure tests are included)
- [ ] This PR requires to update the documentation [n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y]
- [ ] This PR changes the current API [n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by fchollet** [comment]
Tests are failing, please take a look.

**Comment by roywei** [comment]
@fchollet fixed test failure for theano backend. Open and closed few times because some other tests in TF backend hangs. Now all tests are passing.

File: keras/metrics.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def categorical_accuracy(y_true, y_pred):
 
 
 def sparse_categorical_accuracy(y_true, y_pred):
-    return K.cast(K.equal(K.max(y_true, axis=-1),
+    return K.cast(K.equal(K.flatten(y_true),
```
Comment (pavithrasv) [inline keras/metrics.py]: When is this required?

Comment (roywei) [reply keras/metrics.py]: This is just to make sure y_true passed in is in the correct shape. (e.g. y_true = [0, 1, 1] in my example). However I found a unit test [here](https://github.com/keras-team/keras/blob/master/tests/keras/legacy/interface_test.py#L846), it's passing y_true in the shape of `(number of samples, 1)`. so it's like `y_true=[[0], [1], [1]]`. 

I didn't find any documentation specifying the required label shape. So I assumed both are OK, hence added flatten. If this is not the case, please advice.

Comment (pavithrasv) [reply keras/metrics.py]: Got it, thank you. Can we add a small comment explaining that?

**Pull request merged**